examples/ui/size_constraints.rs

```diff
@@ -302,28 +298,24 @@ fn update_buttons(
     mut bar_query: Query<&mut Style, With<Bar>>,
     mut text_query: Query<&mut Text>,
     children_query: Query<&Children>,
-    mut button_activated_event: EventWriter<ButtonActivatedEvent>,
 ) {
     let mut style = bar_query.single_mut();
     for (button_id, interaction, constraint, value) in button_query.iter_mut() {
         match interaction {
-            Interaction::Pressed => {
-                button_activated_event.send(ButtonActivatedEvent(button_id));
-                match constraint {
-                    Constraint::FlexBasis => {
-                        style.flex_basis = value.0;
-                    }
-                    Constraint::Width => {
-                        style.width = value.0;
-                    }
-                    Constraint::MinWidth => {
-                        style.min_width = value.0;
-                    }
-                    Constraint::MaxWidth => {
-                        style.max_width = value.0;
-                    }
+            Interaction::Pressed => match constraint {
+                Constraint::FlexBasis => {
+                    style.flex_basis = value.0;
```

Comment:
The style changes are still happening on "press" but the buttons are only updating on "click."

It is possible to put the UI into an inconsistent state by clicking and holding a radio button, moving the mouse away from the button, and releasing.

Reply:
Good catch, I missed that `update_buttons` also contains button logic.
Fixed it by moving into system that handles clicks and renamed systems to clarify the intention.